### PR TITLE
AR Analysis Template

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Adds and implements a new analysis template to AR. It also adds a new analysis byline component and story and changes the background colour of tags on analysis to white for accessibility.

## Why?
This updates the app analysis design in line with new editorial designs. Design.Analysis previously used the standard layout but the new designs require a reordering of components with some non-standard components used.

## Screenshots

wide
| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before-wide]: https://user-images.githubusercontent.com/20416599/185168695-cfbf81d8-a589-496c-833e-6bf1ed39da87.png
[after-wide]: https://user-images.githubusercontent.com/20416599/185168768-9465b206-01bf-4847-bed6-fa6edbdb791a.png

mobile
| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://user-images.githubusercontent.com/20416599/185168887-b6225c4c-8b94-4eff-a427-14c115e2c97e.png
[after]: https://user-images.githubusercontent.com/20416599/185168931-54e84bd7-0ce5-4ea0-a44d-2f71c5706a5c.png

dark mode -  AR
| Before      | After      |
|-------------|------------|
| ![before-dark][] | ![after-dark][] |

[before-dark]:https://user-images.githubusercontent.com/20416599/185590338-ca900feb-a8df-4a6d-bbe5-e44377725eae.jpeg
[after-dark]: https://user-images.githubusercontent.com/20416599/185590391-f71e5cfb-109a-4699-b2ad-37fab5792f52.png

### CODE REVIEW

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:14]:
```suggestion
const styles = css`
```

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:33]:
```suggestion
const anchorStyles = (format: ArticleFormat) => css`
```

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:29]: Can we use the Source API for this?

```suggestion
	${headline.medium({
		fontWeight: 'light',
		fontStyle: 'italic',
		lineHeight: 'tight',
	})}

	color: ${neutral[46]};
	display: flex;
	flex-direction: column;

	${until.mobileMedium} {
		${headline.small({
			fontWeight: 'light',
			fontStyle: 'italic',
			lineHeight: 'tight',
		})}
	}
```

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:19]: Can we add this to `editorialPalette` as `text.byline`?

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:56]: The `css` function isn't required if you just have a single `SerializedStyles`. It's used for composing multiple sets of styles together; for example:

```ts
const compoundStyles = css(stylesOne, stylesTwo, stylesThree);
```

```suggestion
		styles={styles}
		anchorStyles={anchorStyles(format)}
```

File: apps-rendering/src/components/Layout/AnalysisLayout.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Layout/AnalysisLayout.tsx:77]: Can I suggest pulling the related content and footer outside of the `main` tag? I think the article is the "dominant content" or "central topic" of the page [^1], and the onward content and footer probably fall outside that?

The `ImmersiveLayout` may provide an example of how this could look?

https://github.com/guardian/dotcom-rendering/blob/63733237a4b526b4238c09ec21280a6310a81bc5/apps-rendering/src/components/Layout/ImmersiveLayout.tsx#L83-L108

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main

File: apps-rendering/src/item.ts
Comment (JamieB-gu) [inline apps-rendering/src/item.ts:392]: As far as I can tell nothing has changed here? Is this diff all indentation? 🤔

File: apps-rendering/src/item.ts
Comment (JamieB-gu) [inline apps-rendering/src/item.ts:508]: Are there code changes here, or is this all indentation? If this and the above change to `fromCapiLiveBlog` are both just indentation would it be ok if you reverted them? I think it might make this PR easier to review and may cause less confusion for anyone digging through the blame for this file in future.

**Review by JamieB-gu — COMMENTED** [review]
Nice one integrating with the new sub-component/index/defaults patterns for these new components! I'd be interested to hear your thoughts on this pattern and how easy it is to work with.

A few questions about colour contrast:

1. Do we need to update the colour of the tags against this new background colour?
2. Do we need a dark mode colour for that new "Analysis" line in the standfirst?
3. Does the byline link colour in dark mode meet accessibility guidelines? Firefox is reporting low contrast for that red on black (I haven't been able to test the variants for the other pillars). cc @HarryFischer on this one.

| Tags | Dark Mode |
| - | - |
| ![analysis-tags] | ![analysis-dark] |

[analysis-tags]: https://user-images.githubusercontent.com/53781962/185185262-25d1f04a-8225-4d81-83ef-147914792aca.jpg
[analysis-dark]: https://user-images.githubusercontent.com/53781962/185185271-1878b657-5856-4c1d-a96c-abaed6bf6784.jpg

File: apps-rendering/src/components/Layout/AnalysisLayout.tsx
Comment (abeddow91) [reply apps-rendering/src/components/Layout/AnalysisLayout.tsx:77]: Makes sense, I'll amend this!

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (abeddow91) [reply apps-rendering/src/components/Byline/AnalysisByline.tsx:21]: Definitely, will move this up.

File: apps-rendering/src/item.ts
Comment (abeddow91) [reply apps-rendering/src/item.ts]: Yep all whitespace changes - I've reverted this as requested.

File: apps-rendering/src/item.ts
Comment (abeddow91) [reply apps-rendering/src/item.ts]: Just whitespace changes. I've reverted these and the ones further up so it doesn't muddy the water!

**Review by JamieB-gu — APPROVED** [review]
Looks good, thanks!

Given that you're merging to a feature branch, perhaps we can merge this and make the contrast/accessibility issue a card on your board that we deal with before that branch gets merged to `main`? That way we get these changes in to unblock other work, but we also make sure we fix any accessibility problems before anything gets to production?

**Comment by HarryFischer** [comment]
For me news.500 / #FF5943 (which we currently use in dark mode news articles) passed on the grey background neutral.10 / #1A1A1A. Is that what Firefox is complaining about?

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (JamieB-gu) [inline apps-rendering/src/components/Byline/AnalysisByline.tsx:37]: I've just realised these are using `background.byline` when I think they should probably be using `text.bylineAnchor` and `text.bylineAnchorDark`?

File: common-rendering/src/editorialPalette/background.ts
Comment (JamieB-gu) [inline common-rendering/src/editorialPalette/background.ts:643]: I think you may instead want to add these to `text.bylineAnchor` and `text.bylineAnchorDark`?

**Review by JamieB-gu — COMMENTED** [review]
> For me news.500 / #FF5943 (which we currently use in dark mode news articles) passed on the grey background neutral.10 / #1A1A1A. Is that what Firefox is complaining about?

Ah, I think this is actually using news.300 / AB0613? If you look at the screenshot above the byline red looks quite different to the bullet points below, which are in the more orangey news.500.

I've added a couple of comments below which I think may cover this.

File: common-rendering/src/editorialPalette/background.ts
Comment (abeddow91) [reply common-rendering/src/editorialPalette/background.ts:643]: Good spot, thanks! I'll amend this

File: apps-rendering/src/components/Byline/AnalysisByline.tsx
Comment (abeddow91) [reply apps-rendering/src/components/Byline/AnalysisByline.tsx:37]: Totally right, I'll change this

**Comment by abeddow91** [comment]
@JamieB-gu I've updated this for darkmode using the text palette. I've also updated the background colour of tags in this PR  as it was a small tweak. If you could take another look and see if you're happy that would be 👍

**Review by JamieB-gu — APPROVED** [review]

**Pull request merged**
